msgpack/encode.ts

```diff
@@ -0,0 +1,267 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { BytesList } from "../bytes/bytes_list.ts";
+
+export type EncodeType =
+  | number
+  | bigint
+  | string
+  | boolean
+  | null
+  | Uint8Array
+  | EncodeType[]
+  | EncodeMap;
+
+interface EncodeMap {
+  [index: string | number]: EncodeType;
+}
+
+const FOUR_BITS = 16;
+const FIVE_BITS = 32;
+const SEVEN_BITS = 128;
+const EIGHT_BITS = 256;
+const FIFTEEN_BITS = 32768;
+const SIXTEEN_BITS = 65536;
+const THIRTY_ONE_BITS = 2147483648;
+const THIRTY_TWO_BITS = 4294967296;
+const SIXTY_THREE_BITS = 9223372036854775808n;
+const SIXTY_FOUR_BITS = 18446744073709551616n;
+
+const encoder = new TextEncoder();
+
+export function encode(object: EncodeType) {
+  const byteList = new BytesList();
+  encodeSlice(object, byteList);
+  return byteList.concat();
+}
+
+function encodeNumber(num: number) {
+  if (!Number.isInteger(num)) { // float 64
+    const dataView = new DataView(new ArrayBuffer(9));
+    dataView.setFloat64(1, num);
+    dataView.setUint8(0, 0xcb);
+    return new Uint8Array(dataView.buffer);
+  }
+
+  if (num < 0) {
+    if (num <= -1 && num >= -FIVE_BITS) { // negative fixint
+      return new Uint8Array([num]);
+    }
+
+    if (num >= -SEVEN_BITS) { // int 8
```

Comment:
Concerned about performance, since this implementation is already not the fastest. I think this is pretty readable but if you have a strong opinion the other way, I'd be fine to change this to an absolute value.

Reply:
Nah, it's fine.